one/src/network.rs

```diff
@@ -110,12 +112,54 @@ impl<S> Ipfs<S> {
     pub fn api(&self) -> IpfsMetricsMiddleware<Arc<IpfsService<S>>> {
         self.api.clone()
     }
+    pub fn client(&self) -> IpfsClient {
+        IpfsClient(self.client.clone())
+    }
     pub async fn stop(self) -> Result<()> {
         self.p2p.stop().await?;
         Ok(())
     }
 }
 
+#[derive(Clone)]
+pub struct IpfsClient(P2pClient);
+
+#[async_trait]
+impl ceramic_api::P2PService for IpfsClient {
+    async fn peers(&self) -> Result<Vec<ceramic_api::PeerInfo>> {
+        self.0
+            .get_peers()
+            .await?
+            .into_iter()
+            .map(|(peer_id, addresses)| {
+                Ok(ceramic_api::PeerInfo {
+                    id: NodeId::try_from_peer_id(&peer_id)?,
+                    addresses,
+                })
+            })
+            .collect()
+    }
+    async fn peer_connect(&self, addrs: &[ceramic_api::Multiaddr]) -> Result<()> {
+        // Find peer id in one of the addresses
```

Comment:
is it possible to make a weird request were you specify addresses with multiple peer IDs? it seems like we'd end up replacing the ID in following addresses or am I misunderstanding?

Reply:
hmm good point, we should check all peers ids are the same. As written it would likely just result in that address failing to connect but better to catch the error sooner.